### PR TITLE
Update packaging to comply with standards

### PR DESCRIPTION
**Pull request by cmcantalupo**

### CODE REVIEW

**Comment by bgeltz** [comment]
It would be nice if we could integrate the sphinx docs build into the new build recipe.  It's difficult to deal with in spack when they're part of the regular autotools build, but I need to package geopmdpy separately (as a Python package) to resolve Python version issues between what's in spack vs. the host system.

File: service/setup.py [hunk 1]
```diff
@@ -1,61 +1,7 @@
...
-               'Programming Language :: Python :: 3',
-               'Programming Language :: Python :: 3.6',
-]
-
-install_requires = ['cffi>=1.14.5',
-                    'setuptools>=53.0.0',
-                    'psutil>=5.8.0',
-                    'dasbus>=1.6.0',
-                    'jsonschema>=3.2.0']
+import setuptools
```
Comment (cmcantalupo) [inline service/setup.py]: comment why this is here (debian support)

File: geopmdpy/requirements.txt
Comment (dannosliwcd) [inline geopmdpy/requirements.txt:1]: Add `build` (used by autogen.sh)

File: geopmpy/requirements.txt
Comment (dannosliwcd) [inline geopmpy/requirements.txt:1]: Add `build` (used by autogen.sh)

File: geopmdpy/geopmdpy/__init__.py [hunk 2]
```diff
@@ -1,26 +0,0 @@
...
-#
-
-"""The geopm daemon python package
-
-"""
-
-import os
-
-try:
-    from geopmdpy.version import __version__
```
Comment (cmcantalupo) [inline geopmdpy/geopmdpy/__init__.py]: i guess we may need to add this __init__ file back...

File: geopmdpy/debian/changelog [hunk 3]
```diff
@@ -1,5 +1,5 @@
-geopmdpy (3.0.1.dev222+gd00ee89a4.d20240328-1) UNRELEASED; urgency=low
+python3-geopmdpy (3.0.1.dev222+gd00ee89a4.d20240328-1) UNRELEASED; urgency=low
 
-  * Initial release. Closes: #1055130
```
Comment (cmcantalupo) [inline geopmdpy/debian/changelog]: closes line is handy

**Review by bgeltz — APPROVED** [review]

**Review by dannosliwcd — APPROVED** [review]

**Pull request merged**